### PR TITLE
Replace DKMS guide with a new basic set up guide.

### PR DESCRIPTION
**Pull request by AdityaGarg8**

### CODE REVIEW

**Comment by AdityaGarg8** [comment]
@Redecorating, can you check this out. The pr is for the preinstall branch btw to ensure that things are synced.

File: docs/guides/setting-up.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```sh
+echo apple-bce | sudo tee /etc/modules-load.d/t2.conf
+```
+
+## Make modules load on early boot
+
+Having the `apple-bce` module loaded early allows the use of the keyboard for decrypting encrypted volumes (LUKS).
+It also is useful when boot doesn't work, and the keyboard is required for debugging.
+To do this, one must ensure the `apple-bce` module *as well as its dependent modules* are included in the initial ram disk.
+If your distro uses `initramfs-tools` (all debian-based distros), then `/etc/initramfs-tools/modules` stores a list of extra modules to be included and loaded at early boot time.
```
Comment (Redecorating) [inline docs/guides/setting-up.md:54]: for this section maybe we can do it as a list?

- on systems with initramfs-tools
    1. step 1
    2. step 2
- systems with mkinitcpio
    1. step 1
    2. step 2
- systems with other initramfs/initrd generation systems
    1. refer to the documentation and ensure the kernel module apple-bce is loaded early

or something like that.

also note that not all arch systems use mkinitcpio, some users may choose to use booster or other ones and will have to consult the respective documentation

File: docs/guides/setting-up.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   ```
+  
+- Boot into the [macOS Recovery](https://support.apple.com/en-gb/HT201314) and then restart into Linux.
+- Unplug all the external USB keyboards and mouse and then restart into Linux, keeping them unplugged.
+
+If you still face an issue, mention it [here](https://github.com/t2linux/wiki/issues) or on the discord.
+
+## Fixing Suspend
+
+Copy [this script](../tools/rmmod_tb.sh) to `/lib/systemd/system-sleep/rmmod_tb.sh`
```
Comment (Redecorating) [inline docs/guides/setting-up.md:99]: if the script is not already there I suppose?

File: docs/guides/setting-up.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Now run :-
+
+```sh
+sudo chmod 755 /lib/systemd/system-sleep/rmmod_tb.sh
+sudo chown root:root /lib/systemd/system-sleep/rmmod_tb.sh
+```
+
+Change the path to `/lib64/elogind/system-sleep/rmmod_tb.sh` if using OpenRC on Gentoo as mentioned previously.
+
+It unloads the Touchbar modules as they can cause issues for suspend.
```
Comment (Redecorating) [inline docs/guides/setting-up.md:112]: and reloads them after resume

File: docs/roadmap.md [hunk 4]
```diff
@@ -16,8 +16,9 @@ Still in case something goes wrong we are not responsible for it, we will try ou
 
 While its technically possible to remove macOS entirely, its strongly encouraged to dual boot it with Linux because :-
 
-1. It acts as a backup in case something goes wrong.
-2. macOS updates often bring along certain firmware updates, which tend to be useful for Linux as well.
+1. To set up Wi-Fi on Linux, you will need Wi-Fi firmware, which can be obtained only from macOS.
```
Comment (Redecorating) [inline docs/roadmap.md:19]: maybe say "can be only legally obtained"? to provide a little context as to why the files can't be hosted on a website

File: mkdocs.yml [hunk 5]
```diff
@@ -50,8 +51,8 @@ nav:
           - Notable Contributors: notable-contributors.md
     - Guides:
           - Pre-Install: guides/preinstall.md
+          - Basic setup: guides/setting-up.md
```
Comment (Redecorating) [inline mkdocs.yml:54]: possibly consider "postinstall" as the name, but "setting up" is fine too

File: docs/guides/setting-up.md (hunk 2 above)
Comment (AdityaGarg8) [reply docs/guides/setting-up.md:99]: Wdym?

Comment (Redecorating) [reply docs/guides/setting-up.md:99]: iirc, some t2 specific iso's put that script there already? although it might be an outdated copy of it

Comment (AdityaGarg8) [reply docs/guides/setting-up.md:99]: I don’t think it causes any functional change if the user follows it again. Moreover the guide initially has a Do I need to follow this section.

File: mkdocs.yml (hunk 5 above)
Comment (AdityaGarg8) [reply mkdocs.yml]: Done

File: docs/roadmap.md (hunk 4 above)
Comment (AdityaGarg8) [reply docs/roadmap.md]: Done

File: docs/guides/setting-up.md (hunk 3 above)
Comment (AdityaGarg8) [reply docs/guides/setting-up.md]: Done

File: docs/guides/setting-up.md (hunk 1 above)
Comment (AdityaGarg8) [reply docs/guides/setting-up.md]: Done

**Comment by AdityaGarg8** [comment]
@Redecorating , check now.

File: docs/guides/postinstall.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+```sh
+echo apple-bce | sudo tee /etc/modules-load.d/t2.conf
+```
+
+## Make modules load on early boot
+
+Having the `apple-bce` module loaded early allows the use of the keyboard for decrypting encrypted volumes (LUKS).
+It also is useful when boot doesn't work, and the keyboard is required for debugging.
+To do this, one must ensure the `apple-bce` module *as well as its dependent modules* are included in the initial ram disk.
```
[the remark this answers is not in the document]
Comment (AdityaGarg8) [reply docs/guides/postinstall.md:53]: I'll merge this first to the preinstall branch. Rest editing can be done there.

**Pull request merged**